src/v2/cookbook/electron-vue-cookbook.md

```diff
@@ -0,0 +1,704 @@
+  data() {
+    return {
+      minutes: 1,
+      seconds: 0
+    };
+  }
+};
+</script>
+```
+
+You should see the below:
+
+![Screenshot](https://i.imgur.com/rlQiEST.png)
+
+Our timer looks weird because the minutes and seconds are single digits. We can fix this by adding a computed property that adds zero padding to the minutes and seconds. Make the changes indicated in the comments below.
+
+```vue
+<template>
+<div>
+  <!-- note the change below -->
+  <h1>{{minutesText}}:{{secondsText}}</h1>
+</div>
+</template>
+<script>
+export default {
+  name: "App",
+  data() {
+    return {
+      minutes: 1,
+      seconds: 0
+    };
+  },
+  // add the computed properties
+  computed: {
+    minutesText() {
+      return String(this.minutes).padStart(2, "0");
+    },
+    secondsText() {
+      return String(this.seconds).padStart(2, "0");
+    }
+  }
+};
+</script>
+```
+
+Now our timer looks better.
+
+![Screenshot](https://i.imgur.com/0kKhtVp.png)
+
+Next, let's add a start and reset button for our timer. Make the changes indicated in the comments below.
+
+```vue
+<template>
+<div>
+  <h1>{{minutesText}}:{{secondsText}}</h1>
+
+  <!--add the buttons below-->
+  <button @click="startTimer">Start</button>
+  <button @click="resetTimer">Reset</button>
+</div>
+</template>
+<script>
+export default {
+  name: "App",
+  data() {
+    return {
+      minutes: 1,
+      seconds: 0,
+      // add the timer variable
+      timer: null
+    };
+  },
+  // add the methods
+  methods: {
+    startTimer() {
+      this.timer = setInterval(() => this.countdown(), 1000);
+    },
+    resetTimer() {
+      clearInterval(this.timer);
+      this.timer = null;
+      this.minutes = 1;
+      this.seconds = 0;
+    },
+    countdown() {
+      this.seconds--;
+      if (this.seconds < 0) {
+        this.minutes--;
+        if (this.minutes < 0) {
+          this.resetTimer();
+        } else {
+          this.seconds = 59;
+        }
+      }
+    }
+  },
+  computed: {
+    minutesText() {
+      return String(this.minutes).padStart(2, "0");
+    },
+    secondsText() {
+      return String(this.seconds).padStart(2, "0");
+    }
+  }
+};
+</script>
+```
+
+Now you should be able to start and reset the timer. You might notice the timer counts down faster when you click Start several times while the timer is already running. To avoid this from happening, let's disable the Start button when the timer is already running. We'll add a variable called `running` which will represent the state of the timer and use that variable to disable the Start button while the timer is running. Make the necessary changes indicated in the comments below.
+
+```vue
+<template>
+<div>
+  
+  <h1>{{minutesText}}:{{secondsText}}</h1>
+
+  <!--Use the running variable to disable buttons-->
+  <button @click="startTimer" :disabled="running">Start</button>
+  <button @click="resetTimer" :disabled="!running">Reset</button>
+
+</div>    
+</template>
+
+<script>
+export default {
```

Comment:
IMHO this code example is a bit too long, which makes it harder for 1) readers to follow and 2) maintainers to sync between the markdown and the raw scripts. Can we use something shorter e.g. (OTOMH) the todo app?

Reply:
I considered the todo app but went with the timer because building a timer 
* is quick to build and easy to understand what the code is doing 
* is a small amount of code (and less code than results for todo lists built with Vue on GitHub)

A timer lets us focus on the Electron part of the cookbook. I think with a todo list I'd be spending most of the tutorial on how to build a todo list in Vue, and many existing tutorials for Vue involve building todo lists.

Reply:
I still maintain that the current code is too long for a cookbook. I personally find it hard to follow, and would like to think that defeats the purpose. 
Btw (and I missed this before), it would be far better IMHO if we don't need to rely on a third party boilerplate (simulatedgreg/electron-vue) for this.

Reply:
I agree with @phanan, I think this cookbook entry is valuable but it strays from the point sometimes. In this cookbook you probably shouldn't be explaining computed properties and refactoring. It would be better to show  a bit of the code, and have the full thing hosted elsewhere, and then build on that by only what's changed with one line for context and an ellipsis [...] to convey there's more

Otherwise, you would have to go the other route with a much smaller example. You can make a really small [Vue to do](https://codepen.io/yuugou727/pen/NpQpor) that's more immediately legible than the timer.